app/containers/gistEditorForm/meta.js

```diff
@@ -0,0 +1,153 @@
+  {name: "SPARQL", mime: "application/sparql-query", mode: "sparql", ext: ["rq", "sparql"], alias: ["sparul"]},
+  {name: "Spreadsheet", mime: "text/x-spreadsheet", mode: "spreadsheet", alias: ["excel", "formula"]},
+  {name: "SQL", mime: "text/x-sql", mode: "sql", ext: ["sql"]},
+  {name: "Squirrel", mime: "text/x-squirrel", mode: "clike", ext: ["nut"]},
+  {name: "Stylus", mime: "text/x-styl", mode: "stylus", ext: ["styl"]},
+  {name: "Swift", mime: "text/x-swift", mode: "swift", ext: ["swift"]},
+  {name: "sTeX", mime: "text/x-stex", mode: "stex"},
+  {name: "LaTeX", mime: "text/x-latex", mode: "stex", ext: ["text", "ltx"], alias: ["tex"]},
+  {name: "SystemVerilog", mime: "text/x-systemverilog", mode: "verilog", ext: ["v"]},
+  {name: "Tcl", mime: "text/x-tcl", mode: "tcl", ext: ["tcl"]},
+  {name: "Textile", mime: "text/x-textile", mode: "textile", ext: ["textile"]},
+  {name: "TiddlyWiki ", mime: "text/x-tiddlywiki", mode: "tiddlywiki"},
+  {name: "Tiki wiki", mime: "text/tiki", mode: "tiki"},
+  {name: "TOML", mime: "text/x-toml", mode: "toml", ext: ["toml"]},
+  {name: "Tornado", mime: "text/x-tornado", mode: "tornado"},
+  {name: "troff", mime: "text/troff", mode: "troff", ext: ["1", "2", "3", "4", "5", "6", "7", "8", "9"]},
+  {name: "TTCN", mime: "text/x-ttcn", mode: "ttcn", ext: ["ttcn", "ttcn3", "ttcnpp"]},
+  {name: "TTCN_CFG", mime: "text/x-ttcn-cfg", mode: "ttcn-cfg", ext: ["cfg"]},
+  {name: "Turtle", mime: "text/turtle", mode: "turtle", ext: ["ttl"]},
+  {name: "TypeScript", mime: "application/typescript", mode: "javascript", ext: ["ts"], alias: ["ts"]},
+  {name: "Twig", mime: "text/x-twig", mode: "twig"},
+  {name: "Web IDL", mime: "text/x-webidl", mode: "webidl", ext: ["webidl"]},
+  {name: "VB.NET", mime: "text/x-vb", mode: "vb", ext: ["vb"]},
+  {name: "VBScript", mime: "text/vbscript", mode: "vbscript", ext: ["vbs"]},
+  {name: "Velocity", mime: "text/velocity", mode: "velocity", ext: ["vtl"]},
+  {name: "Verilog", mime: "text/x-verilog", mode: "verilog", ext: ["v"]},
+  {name: "VHDL", mime: "text/x-vhdl", mode: "vhdl", ext: ["vhd", "vhdl"]},
+  {name: "Vue.js Component", mimes: ["script/x-vue", "text/x-vue"], mode: "vue", ext: ["vue"]},
+  {name: "XML", mimes: ["application/xml", "text/xml"], mode: "xml", ext: ["xml", "xsl", "xsd", "svg"], alias: ["rss", "wsdl", "xsd"]},
+  {name: "XQuery", mime: "application/xquery", mode: "xquery", ext: ["xy", "xquery"]},
+  {name: "Yacas", mime: "text/x-yacas", mode: "yacas", ext: ["ys"]},
+  {name: "YAML", mimes: ["text/x-yaml", "text/yaml"], mode: "yaml", ext: ["yaml", "yml"], alias: ["yml"]},
+  {name: "Z80", mime: "text/x-z80", mode: "z80", ext: ["z80"]},
+  {name: "mscgen", mime: "text/x-mscgen", mode: "mscgen", ext: ["mscgen", "mscin", "msc"]},
+  {name: "xu", mime: "text/x-xu", mode: "mscgen", ext: ["xu"]},
+  {name: "msgenny", mime: "text/x-msgenny", mode: "mscgen", ext: ["msgenny"]}
+];
```

Comment:
The `modeInfo` comes from `codemirror/mode/meta.js`. There is also a method called `CodeMirror.findModeByFileName`, which is designed to find mode according to filename. But it can not be called in `index.js` in my try.